pymc3/distributions/transforms.py

```diff
@@ -76,6 +76,9 @@ def __init__(self, dist, transform, *args, **kwargs):
             # force the last dim not broadcastable
             self.type = tt.TensorType(v.dtype, b)
 
+    def random(self, point=None, size=None, repeat=None):
+        return self.transform_used.forward(self.dist.random(point, size, repeat)).eval()
+
```

Comment:
This is awesome! @hvasbath we can get rid of the non-transform restriction with SMC as well with this.

However, currently this is causing a test fail in pymc3/tests/test_random.py::TestDrawValues::test_dep_vars.  The test is to ensure evaluating with transformed vars as input (eg, `sd_log__` instead of `sd`). This does not apply anymore after this, but I am not sure what is the best way to remove that test. Thoughts? @ColCarroll @aseyboldt

Reply:
Update: tested on a mixed model, seems this method is too slow.